MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs

```diff
@@ -82,5 +94,42 @@ public bool IsImageFetcherEnabled(BaseItem baseItem, LibraryOptions libraryOptio
 
             return itemConfig == null || !itemConfig.DisabledImageFetchers.Contains(name, StringComparer.OrdinalIgnoreCase);
         }
+
+        /// <summary>
+        /// Called when the configuration is updated.
+        /// It will refresh the metadata throttler if the relevant config changed.
+        /// </summary>
+        private void OnConfigurationUpdated(object sender, EventArgs e)
+        {
+            int newMetadataRefreshConcurrency = GetMetadataRefreshConcurrency();
+            if (_metadataRefreshConcurrency != newMetadataRefreshConcurrency)
+            {
+                _metadataRefreshConcurrency = newMetadataRefreshConcurrency;
+                SetupMetadataThrottler();
```

Comment:
We might see some weird behaviour if this value is changed mid-scan. On-going threads will use the old semaphore and the new threads will use the new one. I don't think assigning it to a variable inside `RunMetadataRefresh` will do much to prevent it.

Reply:
It's not a blocker, but it might be a good idea to fix that in a new PR or at least keep it in mind if people experience weird issues.